tests/test_loaders_dumpers/test_loaders_pydantic.py

```diff
@@ -1,29 +1,57 @@
 from tests.test_loaders_dumpers.environment import env
-from tests.test_loaders_dumpers.loaderdumpertestcase import LoaderDumperTestCase
 from tests.test_loaders_dumpers.models.books_normalized_pydantic import BookSeries
 from tests.test_loaders_dumpers.models.kitchen_sink_pydantic import Dataset
 
 
-class PydanticLoadersUnitTest(LoaderDumperTestCase):
-    env = env
-
-    def test_yaml_loader_single(self):
-        """Load obo_sample.yaml, emit obo_sample_yaml.yaml and compare to obo_sample_output.yaml"""
-        self.loader_test("book_series_lotr.yaml", BookSeries, yaml_loader)
-
-    def test_json_loader(self):
-        """Load obo_sample.json, emit obo_sample_json.yaml and check the results"""
-        self.loader_test("book_series_lotr.json", BookSeries, json_loader)
-
-    def test_yaml_loader_kitchen_sink(self):
-        self.loader_test("kitchen_sink_normalized_inst_01.yaml", Dataset, yaml_loader)
-
-    def test_json_loader_kitchen_sink(self):
-        self.loader_test("kitchen_sink_normalized_inst_01.json", Dataset, json_loader)
-
-
-if __name__ == "__main__":
-    unittest.main()
+def loader_test(filename: str, model: Union[type[YAMLRoot], type[BaseModel]], loader: Loader) -> None:
+    """
+    Test the various permutations of the supplied loader using the input file 'filename' -- both load and loads
+
+    :param filename: un-pathed file name to load
+    :param model: model to load the file name into
+    :param loader: package that contains 'load' and 'loads' operations
+    """
+    metadata = FileInfo()
+    name, typ = filename.rsplit(".", 1)
+    expected_yaml = env.expected_path("load", name + "_" + typ + ".yaml")
+    if issubclass(model, YAMLRoot):
+        python_obj: YAMLRoot = loader.load(filename, model, metadata=metadata, base_dir=env.indir)
+    elif issubclass(model, BaseModel):
+        python_obj: BaseModel = loader.load(filename, model, metadata=metadata, base_dir=env.indir)
+    else:
+        raise TypeError(f"Unknown target class: {model}")
+    env.eval_single_file(expected_yaml, yaml_dumper.dumps(python_obj))
+
+    # Make sure metadata gets filled out properly
+    rel_path = os.path.abspath(os.path.join(test_base.env.cwd, ".."))
+    assert os.path.normpath("tests/test_loaders_dumpers/input") == os.path.normpath(
+        os.path.relpath(metadata.base_path, rel_path)
+    )
+    assert os.path.normpath(f"tests/test_loaders_dumpers/input/{filename}") == os.path.normpath(
+        os.path.relpath(metadata.source_file, rel_path)
+    )
+
+
+@pytest.mark.parametrize(
+    "filename,model,loader",
+    [
+        ("book_series_lotr.yaml", BookSeries, yaml_loader),
+        ("book_series_lotr.json", BookSeries, json_loader),
+        ("kitchen_sink_normalized_inst_01.yaml", Dataset, yaml_loader),
+        ("kitchen_sink_normalized_inst_01.json", Dataset, json_loader),
+    ],
+)
+def test_loader(filename, model, loader):
+    loader_test(filename, model, loader)
```

Comment:
Do you need to have `loader_test` as a separate function or can you just parametrize it directly (with the appropriate test name, of course 🙂 )?

Reply:
You right...  fixed..